girder-dandi-archive/girder_dandi_archive/rest.py

```diff
@@ -69,21 +75,107 @@ def create_dandiset(self, params):
     @describeRoute(
         Description("Get Dandiset").param("identifier", "Dandiset Identifier", paramType="path")
     )
+    @dandiset_identifier
     def get_dandiset(self, identifier, params):
-
-        if not identifier:
-            raise RestException("identifier must not be empty.")
-
-        if not validate_dandiset_identifier(identifier):
-            raise RestException("Invalid Dandiset Identifier")
-
         # Ensure we are only looking for drafts collection child folders.
         drafts = get_or_create_drafts_collection()
         doc = Folder().findOne({"parentId": drafts["_id"], "meta.dandiset.identifier": identifier})
         if not doc:
             raise RestException("No such dandiset found.")
         return doc
 
+    @access.public
+    @describeRoute(
+        Description("Get Dandiset Owners").param(
+            "identifier", "Dandiset Identifier", paramType="path"
+        )
+    )
+    @dandiset_identifier
+    def get_dandiset_owners(self, identifier, params):
+        return get_dandiset_owners(find_dandiset_by_identifier(identifier))
+
+    @access.user
+    @autoDescribeRoute(
+        Description("Add Dandiset Owners")
+        .param("identifier", "Dandiset Identifier", paramType="path")
+        .jsonParam(
+            "owners",
+            "A JSON list of girder users to add as owners.",
+            paramType="body",
+            requireArray=True,
+        )
+    )
+    @dandiset_identifier
+    def add_dandiset_owners(self, identifier, owners, params):
+        dandiset = find_dandiset_by_identifier(identifier)
+        Folder().requireAccess(dandiset, user=self.getCurrentUser(), level=AccessType.ADMIN)
+
+        # Make sure the list doesn't contain duplicates
+        # Only work with admin level users, removing non-admins from the ACL
+        user_id_to_level = {
+            str(user["id"]): user["level"]
+            for user in Folder().getFullAccessList(dandiset)["users"]
+            if user["level"] == AccessType.ADMIN
```

Comment:
Good suggestion. As far as the difference between those two keys, `"id"` is just how the ACL of a folder returns the id of a user, instead of `"_id"` normally found elsewhere.